File: packages/expo-sensors/android/src/main/java/expo/modules/sensors/services/GyroscopeService.java
// Copyright 2015-present 650 Industries. All rights reserved.

package expo.modules.sensors.services;

import android.content.Context;
import android.hardware.Sensor;

import java.util.Collections;
import java.util.List;

import org.unimodules.core.interfaces.InternalModule;

import expo.modules.interfaces.sensors.services.GyroscopeServiceInterface;

public class GyroscopeService extends SubscribableSensorService implements InternalModule, GyroscopeServiceInterface {
  public GyroscopeService(Context context) {
    super(context);
  }

  @Override
  int getSensorType() {
    return Sensor.TYPE_GYROSCOPE;
  }


  @Override
  public List<Class> getExportedInterfaces() {
    return Collections.<Class>singletonList(GyroscopeServiceInterface.class);
  }
}

File: packages/expo-sensors/android/src/main/java/expo/modules/sensors/services/GyroscopeService.kt
// Copyright 2015-present 650 Industries. All rights reserved.
package expo.modules.sensors.services

import android.content.Context
import android.hardware.Sensor
import expo.modules.interfaces.sensors.services.GyroscopeServiceInterface
import expo.modules.core.interfaces.InternalModule

class GyroscopeService(context: Context?) : SubscribableSensorService(context), InternalModule, GyroscopeServiceInterface {
  override val sensorType: Int = Sensor.TYPE_GYROSCOPE

  override fun getExportedInterfaces(): List<Class<*>> {
    return listOf<Class<*>>(GyroscopeServiceInterface::class.java)
  }
}
